criu/seize.c

```diff
@@ -599,6 +599,34 @@ static int freeze_processes(void)
 				goto err;
 			}
 			nanosleep(&req, NULL);
+
+			if (cgroup_v2)
+				continue;
+
+			/* As per older kernel docs (freezer-subsystem.txt before
+			 * the kernel commit ef9fe980c6fcc1821), if FREEZING is seen,
+			 * userspace should either retry or thaw. While current
+			 * kernel cgroup v1 docs no longer mention a need to retry,
+			 * even recent kernels can't reliably freeze a cgroup v1.
+			 *
+			 * Let's keep asking the kernel to freeze from time to time.
+			 * In addition, do occasional thaw/sleep/freeze.
```

Comment:
ould it be possible to add a reference to the implementation in runc?
https://github.com/opencontainers/runc/commit/d1007b08a33ae63bb695fcb3244300c9bba2658a

Reply:
I am not sure why we may need it here...

Reply:
There are 2 or 3 runc commits which change this code, and it's still failing some time. Not sure which commit to refer to...

Reply:
I thought it would be useful to reference the runc implementation here, as if it changes in the future it might be good to apply similar changes in CRIU. Perhaps we could add something like
> This approach is similar to the one used in `libcontainer/cgroups/fs/freezer.go` in runc.